### PR TITLE
x878 - Adding original samples number and work requester to SGP

### PR DESCRIPTION
**Pull request by BenTopping**
Inital work on adding original sample count and work requestor to SGP page

### CODE REVIEW

**Comment by BenTopping** [comment]
@seenanair sectioningPlanningSpec seems to be failing intermittently. I had a pr yesterday where it failed but after re-running the ci it passed

**Comment by seenanair** [comment]
> @seenanair sectioningPlanningSpec seems to be failing intermittently. I had a pr yesterday where it failed but after re-running the ci it passed (thats why the checks have failed on this PR)

It is Cypress issue which randomly breaks in certain test cases which is perfectly fine in running locally. usually rerunning tests will fix it . Chris once looked into this. I will go through that in some point .

File: cypress/integration/pages/workAllocationSpec.ts [hunk 1]
```diff
@@ -19,59 +24,101 @@ describe("Work Allocation", () => {
...
+          cy.findByText("Number of blocks, slides or original samples required").should("not.exist");;
+      })
+
+      it("should not show an error message if Number of slides has a value", () => {
+          cy.get("input[name='numSlides']").type("1");
+          cy.findByRole("button", { name: /Submit/i }).should("be.visible").click();
+          cy.findByText("Number of blocks, slides or original samples required").should("not.exist");;
+      })
+
+      it("should not show an error message if Number of Blocks has a value", () => {
```
Comment (seenanair) [inline cypress/integration/pages/workAllocationSpec.ts:107]: Please change the test case  label to 'Number of Original samples'

File: src/components/workAllocation/WorkAllocation.tsx [hunk 2]
```diff
@@ -189,11 +198,11 @@ export default function WorkAllocation() {
       .label("Cost Code"),
     isRnD: Yup.boolean().required(),
     numBlocks: Yup.number().max(MAX_NUM_BLOCKANDSLIDES),
-    numSlides: Yup.number()
-      .max(MAX_NUM_BLOCKANDSLIDES)
-      .when("numBlocks", (numBlocks: any, schema: any) => {
-        if (!numBlocks) {
-          return schema.required("Number of blocks or slides required");
+    numSlides: Yup.number().max(MAX_NUM_BLOCKANDSLIDES),
+    numOriginalSamples: Yup.number().max(MAX_NUM_BLOCKANDSLIDES)
```
Comment (seenanair) [inline src/components/workAllocation/WorkAllocation.tsx:202]: Just a doubt - Is the max value for original samples same as blocks and slides?

Comment (BenTopping) [reply src/components/workAllocation/WorkAllocation.tsx:202]: Good point. Ill ask Katy.

File: src/types/stan.ts [hunk 3]
```diff
@@ -69,7 +69,7 @@ type GraphQLErrorWithExtensions = {
 export function extractServerErrors(e: ClientError): ServerErrors {
   return {
     message: e.response.errors
-      ?.map(error => error?.message?.match(/^.*\s:\s(.*)$/)?.[1])
```
[the remark this answers is not in the document]
Comment (BenTopping) [reply src/types/stan.ts:72]: Im going to put the matcher back in for this pull request so it doesnt hold up the rest of the work. Ill open a different one for the error matcher after looking into it a bit more.

File: cypress/integration/pages/workAllocationSpec.ts (hunk 1 above)
Comment (seenanair) [reply cypress/integration/pages/workAllocationSpec.ts:107]: Always happens to me as well

File: src/types/stan.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (seenanair) [reply src/types/stan.ts:72]: That sounds good 👍

**Pull request merged**